manifests/server/config.pp

```diff
@@ -29,16 +29,35 @@
     mode   => '0755',
   }
 
-  file { $cfg_file:
-    ensure  => present,
-    owner   => $owner,
-    group   => $group,
-    mode    => '0644',
-    require => [
-      File[$cfg_dir],
-      Class['dns::server::install']
-    ],
-    notify  => Class['dns::server::service'],
+  case $::osfamily {
+    'RedHat': {
+      file { $cfg_file:
+        ensure  => present,
+        owner   => $owner,
+        group   => $group,
+        mode    => '0644',
+        content => template("${module_name}/named.conf.erb"),
```

Comment:
Mr. Crowley:  Your `dns__server__config_spec.rb` changes introduced a couple of typos:

On line 38, you need to add an additional `}` at the end of the line to close out the `it { …` block.
On line 39, `shoud` needs to be `should`, and I believe you need to enclose the should expression in `{…}` (e.g. `it { should contain_file… }`)